src/plugins/auth/auth.ts

```diff
@@ -205,6 +216,48 @@ const plugin: FastifyPluginAsync<AuthPluginOptions> = async (fastify, options) =
       .catch((err) => log.warn(err, `mailer failed. link: ${link}`));
   }
 
+  async function verifyCredentials(
+    member: Member,
+    body: { email: string; password: string },
+    challenge?: string,
+  ) {
+    /* let verified is used to store the output of bcrypt.compare() 
+    bcrypt.compare() allows to compare the provided password with a stored hash. 
+    It deduces the salt from the hash and is able to then hash the provided password correctly for comparison
+    if they match, verified is true 
+    if they do not match, verified is false
+    if the user does not has a stored hash, verified is null
+    */
+    let verified;
+    if (member.password === null) {
+      verified = null;
+    } else {
+      verified = bcrypt
+        .compare(body.password, member.password)
+        .then(async (res: boolean) => {
+          if (res) {
```

Comment:
For simplification, I would have 
```
if(!res) {
  return false
}
```
It will avoid to scope the following logic.